sdk/resourcemanager/docs/RBAC.md

```diff
@@ -0,0 +1,117 @@
+    .defineAccessPolicy()
+        .forObjectId(identity.principalId())
+        .allowSecretPermissions(SecretPermissions.GET)
+        .attach()
+    .apply();
+```
+
+For system-assigned managed identity, use e.g. `virtualMachine.systemAssignedManagedServiceIdentityPrincipalId()` in place of `identity.principalId()`.
+
+## Accessing Azure Active Directory
+
+SDK provides limited support for accessing Azure Active Directory, generally only for querying applications, users, groups, and service principals.
+
+This requires additional permission from Azure Active Directory, which can be configured from Azure Active Directory blade in Portal, or by Azure CLI.
+
+Since 2.2.0, SDK switched from [Azure Active Directory Grpah API][aad_graph] to [Microsoft Graph API][microsoft_graph].
+
+Permission required:
+- Access application and service principal: Microsoft Graph, Application permission, Application.Read.All
+- Access user: Microsoft Graph, Application permission, User.Read.All
+- Access group: Microsoft Graph, Application permission, Group.Read.All
+
+Sample code to assign role of contributor to another service principal.
+```java
+azure.accessManagement().roleAssignments().define(UUID.randomUUID().toString())
+    .forServicePrincipal("<service-principal-name>")
+    .withBuiltInRole(BuiltInRole.CONTRIBUTOR)
+    .withSubscriptionScope("<subscriptionId>")
+    .create();
+```
+
+For dedicated Java SDK for Microsoft Graph API, please use [Microsoft Graph SDK for Java](https://github.com/microsoftgraph/msgraph-sdk-java).
+
+## Types of managed identity
+
+There are 2 types of managed identities. Their usage in authenticate is slightly different.
+
+For system-assigned managed identity, one can use it without providing client ID:
+```java
+TokenCredential credential = new ManagedIdentityCredentialBuilder().build();
+```
+
+For user-assigned managed identity, one need to provide a client ID of the identity:
+```java
+TokenCredential credential = new ManagedIdentityCredentialBuilder().clientId("<clientId>").build();
+```
+
+If the user-assigned managed identity is managed by this SDK (`azure-resourcemanager-msi`), the client ID can be found via `Identity.clientId()`.
+
+## Chaining managed identity and service principal in authenticate
+
+For application that deploys to Azure resource, but still need to debug in local machine, one can use a `ChainedTokenCredential` to concatenate 2 credentials.
+
+In below sample code, `ManagedIdentityCredential` will take effect when deployed to Azure resource.
+When debugging in local machine, as fallback, `EnvironmentCredential` will pick up configure from system environment, and authenticate via a different [service principal][service_principal], likely with access only to development resource.
+```java
+TokenCredential credential = new ChainedTokenCredentialBuilder()
+    .addLast(new ManagedIdentityCredentialBuilder().build())
+    .addLast(new EnvironmentCredentialBuilder().build())
+    .build();
+```
```

Comment:
These two are all `azure-identity`. Is is better link to the `azure-identity` docs?